vector/src/app/java/im/vector/push/fcm/troubleshoot/TestFirebaseToken.kt

```diff
@@ -28,24 +31,45 @@ class TestFirebaseToken(val fragment: Fragment) : TroubleshootTest(R.string.sett
 
     override fun perform() {
         status = TestStatus.RUNNING
-        fragment.activity?.let { fragmentActivity ->
-            FirebaseInstanceId.getInstance().instanceId
-                    .addOnCompleteListener(fragmentActivity) { task ->
-                        if (!task.isSuccessful) {
-                            val errorMsg = if (task.exception == null) "Unknown" else task.exception!!.localizedMessage
-                            description = fragment.getString(R.string.settings_troubleshoot_test_fcm_failed, errorMsg)
-                            status = TestStatus.FAILED
-
-                        } else {
-                            task.result?.token?.let {
-                                val tok = it.substring(0, Math.min(8, it.length)) + "********************"
-                                description = fragment.getString(R.string.settings_troubleshoot_test_fcm_success, tok)
-                                Log.e(this::class.java.simpleName, "Retrieved FCM token success [$it].")
+        val activity = fragment.activity
+        if (activity != null) {
+            try {
+                FirebaseInstanceId.getInstance().instanceId
+                        .addOnCompleteListener(activity) { task ->
+                            if (!task.isSuccessful) {
+                                val errorMsg = if (task.exception == null) "Unknown" else task.exception!!.localizedMessage
+                                //Can't find where this constant is (not documented -or deprecated in docs- and all obfuscated)
+                                if ("SERVICE_NOT_AVAILABLE".equals(errorMsg)) {
+                                    description = fragment.getString(R.string.settings_troubleshoot_test_fcm_failed_service_not_available, errorMsg)
+                                } else if ("TOO_MANY_REGISTRATIONS".equals(errorMsg)) {
+                                    description = fragment.getString(R.string.settings_troubleshoot_test_fcm_failed_too_many_registration, errorMsg)
+                                } else if ("ACCOUNT_MISSING".equals(errorMsg)) {
+                                    description = fragment.getString(R.string.settings_troubleshoot_test_fcm_failed_account_missing, errorMsg)
+                                    quickFix = object : TroubleshootQuickFix(R.string.settings_troubleshoot_test_fcm_failed_account_missing_quick_fix) {
+                                        override fun doFix() {
+                                            val intent = Intent(Settings.ACTION_ADD_ACCOUNT)
+                                            intent.putExtra(Settings.EXTRA_ACCOUNT_TYPES, arrayOf("com.google"));
+                                            fragment.startActivityForResult(intent, NotificationTroubleshootTestManager.REQ_CODE_FIX)
```

Comment:
`startActivityForResult()` should be protected by a try catch

Reply:
done